extension/icu/icu-datepart.cpp

```diff
@@ -350,9 +416,16 @@ struct ICUDatePart : public ICUDateFunc {
 						auto &child_entry = child_entries[col];
 						if (is_finite) {
 							FlatVector::Validity(*child_entry).SetValid(i);
-							auto pdata = FlatVector::GetData<int64_t>(*child_entry);
-							auto adapter = info.adapters[col];
-							pdata[i] = adapter(calendar, micros);
+							const auto part_index = size_t(info.part_codes[col]);
+							if (part_index < BEGIN_DOUBLE) {
```

Comment:
Same as before, maybe the switch could be moved to another function and return a bool, then we can easily use it here as well